compiler/semtypinst.nim

```diff
@@ -630,7 +630,10 @@ proc replaceTypeVarsTAux(cl: var TReplTypeVars, t: PType): PType =
       result = t
 
       # Slow path, we have some work to do
-      if result.n != nil and t.kind == tyObject:
```

Comment:
I don't understand this patch. First of all, if ``t.kind == tyRef`` you cannot assume that the child is an object.

Reply:
fixed

Reply:
We alias ``result`` to ``t``. We know ``t.kind == tyRef`` so we set ``result.n`` for ``result.kind == tyRef``. But a ref type has no ``n`` field. This patch cannot be correct.

Reply:
But this works too:

```nim

      if t.kind == tyRef and t.len > 0 and t[0].kind == tyObject and t[0].n != nil:
        discard replaceObjBranches(cl, t[0].n)

```